app/services/invoice_exporter.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+class InvoiceExporter
+  HEADERS = [
+    'Kdnr', 'Name1', 'Name2', 'Strasse', 'Plz', 'Ort',
+    'RG-Nummer', 'Inkl. MWST', 'exkl. MWST', 'ESR-Ref.Nr.'
+  ].freeze
+
+  def initialize(invoices)
+    @invoices = invoices
+  end
+
+  def export(options = {})
+    CSV.generate(csv_options.merge(options)) do |csv|
+      csv << HEADERS
+      @invoices.each { |invoice| csv << invoice_attributes(invoice) }
+    end
+  end
+
+  def csv_options
+    {
+      headers: true, force_quotes: true
+    }
+  end
+
+  # rubocop: disable Metrics/AbcSize
+  def invoice_attributes(invoice)
+    invoice.instance_eval do
```

Comment:
i don't use `instance_eval` that often. what exactly does it?

Reply:
You can find more here: https://lmgtfy.com/?q=ruby+instance_eval

:-P